### PR TITLE
Add Van der Waals equation of state

### PR DESCRIPTION
**Pull request by Glebfra**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by imSanko — APPROVED** [review]
Nice !!

File: physics/vander_vaals_gas_law.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    >>> system_pressure(1, 100, 1, 0.1382, 31.9e-6)
+    831.3345857818664
+    >>> system_pressure(1, 300, -1, 0.1382, 31.9e-6)
+    Traceback (most recent call last):
+        ...
+    ValueError: Please provide the positive values
+    """
+
+    if temperature < 0 or volume < 0:
+        raise ValueError("Please provide the positive values")
```
Comment (cclauss) [inline physics/vander_vaals_gas_law.py:59]:
```suggestion
        raise ValueError("Please provide non-negative values")
```

File: physics/vander_vaals_gas_law.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+CONSTANTS = {
+    "nitrogen": {"a": 0.1370, "b": 38.7e-6},
+    "ammonia": {"a": 0.4225, "b": 37.1e-6},
+    "argon": {"a": 0.1355, "b": 32.0e-6},
+    "oxygen": {"a": 0.1382, "b": 31.9e-6},
+}
+
+
+def system_pressure(
+    quantity: float, temperature: float, volume: float, a: float, b: float
```
Comment (cclauss) [inline physics/vander_vaals_gas_law.py:40]: Please replace `a` and `b` with self-documentating variable names.

Comment (cclauss) [reply physics/vander_vaals_gas_law.py:40]: What happens when quantity is zero?  When quantity is negative?

File: physics/vander_vaals_gas_law.py (hunk 1 above)
Comment (cclauss) [reply physics/vander_vaals_gas_law.py:59]: Please add a test for both values being zero.

File: physics/vander_vaals_gas_law.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    >>> system_temperature(1, 100, 1, 0.1382, 31.9e-6)
+    12.04347294491859
+    >>> system_temperature(1, 300, -1, 0.1382, 31.9e-6)
+    Traceback (most recent call last):
+        ...
+    ValueError: Please provide the positive values
+    """
+
+    if pressure < 0 or volume < 0:
+        raise ValueError("Please provide the positive values")
```
Comment (cclauss) [inline physics/vander_vaals_gas_law.py:83]: Non-negative

File: physics/vander_vaals_gas_law.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    Calculate the critical temperature from two parameters for each gas
+    ---------------------
+    | T_c=8a/(27bR) |
+    ---------------------
+
+    >>> critical_temperature(0.1382, 31.9e-6)
+    154.3865270378366
+    """
+
+    return 8 * a / (27 * b * R)
```
Comment (cclauss) [inline physics/vander_vaals_gas_law.py:98]: Please replace b & R with self-documentating names

**Review by cclauss — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
Closing awaiting_changes PRs to prepare for Hacktoberfest 2024
